standard/attributes.md

```diff
@@ -1060,6 +1060,44 @@ Specifies that a nullable argument won’t be `null` when the method returns the
 >
 > *end example*
 
+### §enumerator-cancellation The EnumeratorCancellation attribute
+
+Specifies the parameter representing the `CancellationToken` for an asynchronous iterator (§15.15). When the body of async iterator method accesses this parameter, it will receive a combination of two cancellation tokens: the token which was actually passed to the method, and the token passed to the `IAsyncEnumerable<T>.GetAsyncEnumerator(CancellationToken)` implementation on the generated iterator. The combined token will be canceled if either of the two source tokens are canceled.
+
+It is an error if the `EnumeratorCancellation` attribute is applied to more than one parameter. The compiler may produce a warning if the `EnumeratorCancellation` attribute is applied to a parameter of a different type than `CancellationToken` or on a method that isn't an asynchronous iterator (§15.15).
+
+> *Example*: The method `GetStringsAsync()` is an asynchronous iterator. Before doing any work to retrieve the next value, it checks the cancellation token to determine if the iteration should be canceled. If cancellation is requested, no further action is taken.
+>
+> <!-- Example: {template:"code-in-class-lib", name:"AsyncEnumeratorCancellation"} -->
+> ```csharp
+> static async IAsyncEnumerable<object> GetStringsAsync([EnumeratorCancellation] CancellationToken token)
+> {
+>     for (int i = 0; i < 10; i++)
+>     {
+>         if (token.IsCancellationRequested)
+>         {
+>             yield break; // Exit if cancellation is requested
+>         }
+>         await Task.Delay(100);
+>         yield return i.ToString();
+>     }
+> }
+> ```
+>
+> The `EnumerationCancellation` attribute instructs the compiler that the associated parameter is used for cancellation. Other cancellation tokens should be composed into a single aggregate token and substituted for the argument to this parameter. In the following example, one token is passed to the `GetStringsAsync` method, and another is passed to the `WithCancellation` extension method. The compiler can create a combined token that requests cancellation when either token has requested cancellation. The async iterator then cancels the operation based on the state of the aggregate token.
```

Comment:
"Combined" and "aggregate" were competing synonyms. "Combined" goes along with the .NET API call that you make to do the combining.

```suggestion
> The `EnumerationCancellation` attribute instructs the compiler that the associated parameter is used for cancellation. Other cancellation tokens should be composed into a single combined token and substituted for the argument to this parameter. In the following example, one token is passed to the `GetStringsAsync` method, and another is passed to the `WithCancellation` extension method. The compiler can create a combined token that requests cancellation when either token has requested cancellation. The async iterator then cancels the operation based on the state of the combined token.
```